core/services/terraform_service.rb

```diff
@@ -24,8 +25,9 @@ def self.apply(resource, logger, path = Dir.pwd)
     ShellCommands.run_command_in_dir(logger, "terraform apply -auto-approve -target=#{resource}", path)
   end
 
-  def self.destroy(resource, logger, path = Dir.pwd)
-    ShellCommands.run_command_in_dir(logger, "terraform destroy -auto-approve -target=#{resource}", path)
+  def self.destroy(resources, logger, path = Dir.pwd)
+    target_args = resources.map { |resource| "-target=#{resource}" }.join(' ')
+    ShellCommands.run_command_in_dir(logger, "terraform destroy -auto-approve #{target_args}", path)
```

Comment:
Please check that Terraform finishes it's operations with the successful return code.